### PR TITLE
#389: add some logging for error of ConnectionPool.tryConnect

### PR DESCRIPTION
**Pull request by Maximilan4**
Description at: #389 
I added some logging for ConnectionPool.tryConnect err result.

### CODE REVIEW

File: pool/connection_pool.go [hunk 1]
```diff
@@ -1417,7 +1425,10 @@ func (p *ConnectionPool) controller(ctx context.Context, e *endpoint) {
 					// Relocate connection between subpools
 					// if ro/rw was updated.
 					if e.conn == nil {
-						p.tryConnect(ctx, e)
+						if err = p.tryConnect(ctx, e); err != nil {
+							log.Printf("tarantool: reopen connection to %s failed: %s\n",
+								e.name, err)
+						}
```
Comment (oleg-jukovec) [inline pool/connection_pool.go:1431]: The method-level scope of the variable is confusing a bit. The local scope is enough.
```suggestion
						if err := p.tryConnect(ctx, e); err != nil {
							log.Printf("tarantool: reopen connection to %s failed: %s\n",
								e.name, err)
						}
```

**Review by oleg-jukovec — CHANGES_REQUESTED** [review]
Just few non-critical notes:

1. Please, fix the review comment.
2. Please, add the changelog entry here:
https://github.com/tarantool/go-tarantool/blob/be0f71e4ede457e9bb7dd05fd4e8fa3db499cfbc/CHANGELOG.md?plain=1#L12

Something like this:
```
## Added
- Log messages if the pool unable to connect to an instance (#390)
```

3. Please, squash commits into one commit and update the commit message. It could be something like:

```
pool: add log messages on connect error

I added some logging for ConnectionPool.tryConnect err result.

Part of #389
```

It will help us to have commits in the same style.

File: pool/connection_pool.go (hunk 1 above)
Comment (Maximilan4) [reply pool/connection_pool.go]: done

**Review by oleg-jukovec — APPROVED** [review]
Thank you for the patch!

**Review by DerekBum — APPROVED** [review]
Lgtm! Thanks for the patch!

**Pull request merged**